File: Hard/Multi String Search.py
def multi_string_search(text, patterns):
    result = []

    for pattern in patterns:
        # Use the built-in find method to check if the pattern is present in the text
        if text.find(pattern) != -1:
            result.append(pattern)

    return result


if __name__ == "__main__":
    # Sample text and patterns
    text = "This is a sample text for multi-string search."
    patterns = ["sample", "search", "notfound"]

    # Perform multi-string search
    result = multi_string_search(text, patterns)

    # Print the found patterns
    print("Found patterns:", *result)

File: Multi String Search.java
import java.util.ArrayList;
import java.util.List;

public class MultiStringSearch {

    // Function to perform multi-string search
    // Returns a list containing patterns found in the text
    public static List<String> multiStringSearch(String text, List<String> patterns) {
        List<String> result = new ArrayList<>();

        // Iterate through each pattern
        for (String pattern : patterns) {
            // Use String.contains to check if the pattern is present in the text
            if (text.contains(pattern)) {
                result.add(pattern);
            }
        }

        return result;
    }

    public static void main(String[] args) {
        // Sample text and patterns
        String text = "This is a sample text for multi-string search.";
        List<String> patterns = List.of("sample", "search", "notfound");

        // Perform multi-string search
        List<String> result = multiStringSearch(text, patterns);

        // Print the found patterns
        System.out.print("Found patterns:");
        for (String pattern : result) {
            System.out.print(" " + pattern);
        }
        System.out.println();
    }
}
